src/api/config.py

```diff
@@ -94,23 +99,120 @@ def _check_config() -> None:
         raise ValueError(error_messages.CONFIG_SIGFIGS_FALLBACK_VALID_RANGE)
 
 
+def _set_default_config() -> None:
+    global configuration  # pylint: disable=global-statement
+
+    configuration = _default_config
+
+
+_default_config = Config(
+    sigfigs=-1,
+    decimal_places=-1,
+    print_auto=False,
+    export_auto_to="",
+    min_exponent_for_non_scientific_notation=-2,
+    max_exponent_for_non_scientific_notation=3,
+    identifier="result",
+    sigfigs_fallback=2,
+    decimal_places_fallback=-1,
+    siunitx_fallback=False,
+    precision=100,
+    ignore_result_overwrite=False,
+)
+
+
 # pylint: disable-next=too-many-arguments
-def config_init(
-    sigfigs: int = -1,  # -1: "per default use rounding rules instead"
-    decimal_places: int = -1,  # -1: "per default use rounding rules instead"
-    print_auto: bool = False,
-    export_auto_to: str = "",
-    min_exponent_for_non_scientific_notation: int = -2,
-    max_exponent_for_non_scientific_notation: int = 3,
-    identifier: str = "result",
-    sigfigs_fallback: int = 2,
-    decimal_places_fallback: int = -1,  # -1: "per default use sigfigs as fallback instead"
-    siunitx_fallback: bool = False,
-    precision: int = 100,
-    ignore_result_overwrite: bool = False,
+def config(
+    sigfigs: Union[
+        int, None, ChangeType
+    ] = ChangeType.NO_CHANGE,  # -1: "per default use rounding rules instead"
+    decimal_places: Union[
+        int, None, ChangeType
+    ] = ChangeType.NO_CHANGE,  # -1: "per default use rounding rules instead"
+    print_auto: Union[bool, None, ChangeType] = ChangeType.NO_CHANGE,
+    export_auto_to: Union[str, None, ChangeType] = ChangeType.NO_CHANGE,
+    min_exponent_for_non_scientific_notation: Union[int, None, ChangeType] = ChangeType.NO_CHANGE,
+    max_exponent_for_non_scientific_notation: Union[int, None, ChangeType] = ChangeType.NO_CHANGE,
+    identifier: Union[str, None, ChangeType] = ChangeType.NO_CHANGE,
+    sigfigs_fallback: Union[int, None, ChangeType] = ChangeType.NO_CHANGE,
+    decimal_places_fallback: Union[
+        int, None, ChangeType
+    ] = ChangeType.NO_CHANGE,  # -1: "per default use sigfigs as fallback instead"
+    siunitx_fallback: Union[bool, None, ChangeType] = ChangeType.NO_CHANGE,
+    precision: Union[int, None, ChangeType] = ChangeType.NO_CHANGE,
+    ignore_result_overwrite: Union[bool, None, ChangeType] = ChangeType.NO_CHANGE,
 ) -> None:
+    """Set the configuration for the application. Pass in the values you want to change. Pass in None for values that you want to reset."""
     global configuration  # pylint: disable=global-statement
 
+    if sigfigs == ChangeType.NO_CHANGE:
+        sigfigs = configuration.sigfigs
+    if sigfigs is None:
+        sigfigs = _default_config.sigfigs
```

Comment:
Cool, haven't thought about doing it that way. Clever!

However: whenever you copy-paste something, you might want to ask yourself if there's not a better way to achieve things. This will make the code look nicer and reduce maintability costs.

Notice that you have the same `if` going on for every single parameter. Please extract this to a separate method, e.g. `set_config_parameter()` or alike.